otx/core/action_dataset_adapter.py

```diff
@@ -0,0 +1,169 @@
+
+logger = get_logger()
+
+class ActionBaseDatasetAdapter(BaseDatasetAdapter):
+    def import_dataset(
+        self,
+        train_data_roots: str,
+        val_data_roots: str,
+        test_data_roots: str = None,
+        unlabeled_data_roots: str = None
+    ) -> DatumaroDataset:
+        """ Import multiple videos that have CVAT format annotation. """
+        self.dataset = {}
+        self.dataset[Subset.TRAINING] = self._prepare_cvat_pair_data(train_data_roots)
+        self.dataset[Subset.VALIDATION] = self._prepare_cvat_pair_data(val_data_roots)
+        if test_data_roots:
+            self.dataset[Subset.TESTING] = self._prepare_cvat_pair_data(test_data_roots)
+
+        return self.dataset
+
+    def _prepare_cvat_pair_data(self, path:str) -> List[DatumaroDataset]:
+        """ Preparing a list of DatumaroDataset. """
+        cvat_data_list = []
+        for cvat_data in os.listdir(path):
+            cvat_data_path = osp.join(path, cvat_data)
+            cvat_data_list.append(DatumaroDataset.import_from(cvat_data_path, 'cvat'))
+        return cvat_data_list
+    
+    def _prepare_label_information(self, datumaro_dataset: dict) -> dict:
+        """ Prepare and reorganize the label information for merging multiple video information.
+        
+        Description w/ examples:
+
+        [Making overall categories]
+        Suppose that video1 has labels=[0, 1, 2] and video2 has labels=[0, 1, 4], 
+        then the overall label should include all label informations as [0, 1, 2, 4].
+
+        [Reindexing the each label index of multiple video datasets]
+        In this case, if the label for 'video1/frame_000.jpg' is 2, then the index of label is set to 2.
+        For the case of video2, if the label for 'video2/frame_000.jpg' is 4, then the index of label is set to 2.
+        However, Since overall labels are [0, 1, 2, 4], 'video2/frame_000.jpg' should has the label index as 3.
+        
+        """
+        outputs = {
+            "category_items": [],
+            "label_groups" : [],
+            "label_entities": [],
+        }
+
+        category_list = [] # to check the duplicate case
+        for cvat_data in datumaro_dataset[Subset.TRAINING]:
+            # Making overall categories
+            categories = cvat_data.categories().get(DatumaroAnnotationType.label, None)
+            
+            if categories not in category_list:
+                outputs["category_items"].extend(categories.items)
+                outputs["label_groups"].extend(categories.label_groups)
+            
+            category_list.append(categories)
+
+            # Reindexing the each label index of multiple video datasets
+            for cvat_data_item in cvat_data:
+                for ann in cvat_data_item.annotations:
+                    ann_name = categories.items[ann.label].name
+                    ann.label = [i for i, category_item in enumerate(outputs["category_items"]) if category_item.name == ann_name][0]
```

Comment:
is this required?

Reply:
Because CVAT could only take a single video. So if there are many videos, unifying labels is needed. (I added detailed description by the docstring)